wgengine/monitor/monitor_linux.go

```diff
@@ -37,18 +36,21 @@ type nlConn struct {
 	buffered []netlink.Message
 }
 
-func newOSMon(logf logger.Logf, _ *Mon) (osMon, error) {
+func newOSMon(logf logger.Logf, m *Mon) (osMon, error) {
 	conn, err := netlink.Dial(unix.NETLINK_ROUTE, &netlink.Config{
 		// Routes get us most of the events of interest, but we need
 		// address as well to cover things like DHCP deciding to give
 		// us a new address upon renewal - routing wouldn't change,
 		// but all reachability would.
 		Groups: unix.RTMGRP_IPV4_IFADDR | unix.RTMGRP_IPV6_IFADDR | unix.RTMGRP_IPV4_ROUTE | unix.RTMGRP_IPV6_ROUTE,
 	})
-	if err != nil {
-		return nil, fmt.Errorf("dialing netlink socket: %v", err)
+	if err == nil {
```

Comment:
it's more common style-wise to use "if err != nil" over "if err == nil". Also, Go's a big fan of returning early and outdenting the rest, making the normal control flow have equal indentation through the function.

Reply:
Done